pkg/utils/etcdutil/etcdutil_test.go

```diff
@@ -537,6 +538,50 @@ func (suite *loopWatcherTestSuite) TestWatcherLoadLimit() {
 	}
 }
 
+func (suite *loopWatcherTestSuite) TestWatcherLoadLargeKey() {
+	re := suite.Require()
+	// use default limit to test 16384 key in etcd
+	count := 16384
+	ctx, cancel := context.WithCancel(suite.ctx)
+	defer cancel()
+	for i := 0; i < count; i++ {
+		suite.put(re, fmt.Sprintf("TestWatcherLoadLargeKey/test-%d", i), "")
+	}
+	cache := struct {
+		syncutil.RWMutex
```

Comment:
Why do we need a lock here?

Reply:
we check result in a goroutine, watcher update cache in another goroutine